packages/style-engine/src/styles/border/index.ts

```diff
@@ -1,31 +1,36 @@
+/**
+ * External dependencies
+ */
+import { camelCase } from 'change-case';
+
 /**
  * Internal dependencies
  */
-import type {
-	BorderIndividualStyles,
-	BorderIndividualProperty,
-	GeneratedCSSRule,
-	Style,
-	StyleDefinition,
-	StyleOptions,
-} from '../../types';
-import { generateRule, generateBoxRules, upperFirst } from '../utils';
+import type { BoxEdge, GenerateFunction, StyleDefinition } from '../../types';
+import { generateRule, generateBoxRules } from '../utils';
+
+function createBorderGenerateFunction( path: string[] ): GenerateFunction {
+	return ( style, options ) =>
+		generateRule( style, options, path, camelCase( path.join( ' ' ) ) );
```

Comment:
I'd say we don't need an external library here. 

There's already a util called `upperFirst`. See the [unit test](https://github.com/WordPress/gutenberg/blob/d265f3030480cda0a390208a880a4d8e96e3c409/packages/style-engine/src/test/utils.js#L7).

So, until we need extra functionality, maybe we could do the following:

```js
const path = [ 'some', 'path', 'to', 'somewhere' ];

console.log( path.map( ( value, index) => index !== 0 ? upperFirst( value ) : value ).join( '' ) );  
// -> somePathToSomewhere
```

Or we could roll our own `camelCase` using `upperFirst()`:

```js
const path = [ 'some', 'path', 'to', 'somewhere' ];

function camelCase( [ firstItem, ...rest ] ) {
	return firstItem.toLowerCase() + rest.map( upperFirst ).join( '' );
}

console.log( camelCase( path ) );
// -> somePathToSomewhere
```

What do you reckon?

Reply:
I was using the lodash version of camelCase first which didn't require adding a dependency, but apparently that is [being phased out](https://github.com/WordPress/gutenberg/pull/43220) elsewhere in Gutenberg in favor of change-case.

But adding our own function is easy enough and it can be trimmed a bit more to our use-case, so that's fixed in 990f0a1.